### ISSUE TITLE
issue when trying to compute EVI2 from a level 1C Sentinel 2B image

### ISSUE THREAD

**Issue by vrgaliano** [ranghetti/sen2r#310]
Hi, I am trying to compute EVI2 index. It works well for level 2A images but not for level 1C. I should be mentioned that the image which is
[SQB_JSON.txt](https://github.com/ranghetti/sen2r/files/4560503/SQB_JSON.txt)
[SQB_JSON.txt](https://github.com/ranghetti/sen2r/files/4560504/SQB_JSON.txt)

 not working is a S2B image that was offline before. However, I could download it but the processing was interrupted and neither the level 2 nor EVI2 were computed.

The JSON file is attached. Please see the message I got below:
```
sen2r()
[2020-04-30 20:22:53] #### Starting sen2r execution. ####
In online mode, at least one parameter among 'extent' and 's2tiles_selected' must be
provided.
[2020-04-30 20:22:53] Launching GUI...

Listening on http://127.0.0.1:5290
Warning: Parameter "apihub" was not specified; setting it to the default ('NA').
[2020-04-30 20:23:31] Gui closed by the user. Starting processing.
[2020-04-30 20:23:31] Searching for available SAFE products on SciHub...
[2020-04-30 20:23:34] Computing output names...
[2020-04-30 20:23:35] Starting to download the required level-2A SAFE products.
No L2A images are needed.
[2020-04-30 20:23:35] Download of level-2A SAFE products terminated.
[2020-04-30 20:23:35] Starting to download the required level-1C SAFE products.
[2020-04-30 20:23:35] Check if products are available for download...
[2020-04-30 20:23:35] 1 Sentinel-2 images are already online.
[2020-04-30 20:23:35] Downloading Sentinel-2 image 1 of 1
         (S2A_MSIL1C_20180805T110621_N0206_R137_T29SQB_20180805T133536.SAFE)...
[2020-04-30 20:24:50] Download of level-1C SAFE products terminated.
[2020-04-30 20:24:50] Updating output names...
+----------------------------------------------------------------------------------------
¦ sen2r Processing Report
¦----------------------------------------------------------------------------------------
¦ Dates to be processed based on processing parameters: 1
¦ Processing completed for: all expected dates.
+----------------------------------------------------------------------------------------
[2020-04-30 20:24:51] #### sen2r session terminated. ####
The processing chain can be re-launched with the command:
  sen2r("C:/Users/usuario/DOCUME~1/SEN2R~1/proc_par/s2proc_20200430_202331.json")
```

**Comment by ranghetti** [comment]
There are some inconsistencies in what you are saying / performing: you are speaking about "a S2B image that was offline before"; but, for what I can see from the two attached JSON files (which are identical), only S2A images (a single date) is available in the specified time window. For this reason, the processing should stop even before the download of the image.

Please attach the two proper JSON files: the one which is working correcly with L2A levels and the one which does not work using L1C. Also, report both the outputs and provide system information (#186).

**Comment by vrgaliano** [comment]
Hi, sorry. Yes, it was S2A this time. I tried with different dates and it happened the same for S2A and S2B. I have attached the correct JSON now. 
> sen2r()
[2020-05-02 11:07:25] #### Starting sen2r execution. ####
In online mode, at least one parameter among 'extent' and 's2tiles_selected' must be
provided.
[2020-05-02 11:07:25] Launching GUI...
Loading required package: shiny

Listening on http://127.0.0.1:3799
Warning: Parameter "apihub" was not specified; setting it to the default ('NA').
Loading required namespace: stringi
[2020-05-02 11:17:24] Gui closed by the user. Starting processing.
[2020-05-02 11:17:24] Searching for available SAFE products on SciHub...
[2020-05-02 11:17:33] Computing output names...
Linking to GEOS 3.6.1, GDAL 2.2.3, PROJ 4.9.3
[2020-05-02 11:17:33] Starting to download the required level-2A SAFE products.
No L2A images are needed.
[2020-05-02 11:17:33] Download of level-2A SAFE products terminated.
[2020-05-02 11:17:33] Starting to download the required level-1C SAFE products.
Images S2A_MSIL1C_20180805T110621_N0206_R137_T29SQB_20200430T042611.SAFE are already on
your system and will be skipped. Set "overwrite_safe" to TRUE to re-download them.
No L1C images are needed.
[2020-05-02 11:17:33] Download of level-1C SAFE products terminated.
[2020-05-02 11:17:33] Starting to correct level-1C SAFE products with Sen2Cor. This
         operation could take very long time.
[2020-05-02 11:17:38] Updating output names...
Error in { : task 1 failed - "task 1 failed - "
This product (S2A_MS~1.SAF) is not in the right format (not recognised).""

> sessionInfo()
R version 4.0.0 (2020-04-24)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18363)

Matrix products: default

locale:
[1] LC_COLLATE=Spanish_Spain.1252  LC_CTYPE=Spanish_Spain.1252    LC_MONETARY=Spanish_Spain.1252
[4] LC_NUMERIC=C                   LC_TIME=Spanish_Spain.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] sf_0.9-2      shiny_1.4.0.2 sen2r_1.3.4  

loaded via a namespace (and not attached):
 [1] httr_1.4.1           shinyFiles_0.8.0     jsonlite_1.6.1       geojsonlint_0.4.0   
 [5] foreach_1.5.0        assertthat_0.2.1     sp_1.4-1             yaml_2.2.1          
 [9] pillar_1.4.3         lattice_0.20-41      glue_1.4.0           digest_0.6.25       
[13] promises_1.1.0       colorspace_1.4-1     htmltools_0.4.0      httpuv_1.5.2        
[17] XML_3.99-0.3         pkgconfig_2.0.3      httpcode_0.3.0       raster_3.1-5        
[21] stars_0.4-1          xtable_1.8-4         scales_1.1.0         later_1.0.0         
[25] mapedit_0.6.0        tibble_3.0.0         leaflet.extras_1.0.0 ellipsis_0.3.0      
[29] shinyjs_1.1          geojsonio_0.9.2      RcppTOML_0.1.6       lazyeval_0.2.2      
[33] cli_2.0.2            magrittr_1.5         crayon_1.3.4         mime_0.9            
[37] maptools_0.9-9       fs_1.4.1             fansi_0.4.1          doParallel_1.0.15   
[41] xml2_1.3.1           lwgeom_0.2-3         foreign_0.8-78       class_7.3-16        
[45] shinydashboard_0.7.1 tools_4.0.0          data.table_1.12.8    lifecycle_0.2.0     
[49] geojson_0.3.2        V8_3.0.2             munsell_0.5.0        compiler_4.0.0      
[53] e1071_1.7-3          jqr_1.1.0            rlang_0.4.5          classInt_0.4-3      
[57] units_0.6-6          grid_4.0.0           iterators_1.0.12     rstudioapi_0.11     
[61] htmlwidgets_1.5.1    crosstalk_1.1.0.1    shinyWidgets_0.5.1   codetools_0.2-16    
[65] jsonvalidate_1.1.0   abind_1.4-5          DBI_1.1.0            curl_4.3            
[69] R6_2.4.1             rgdal_1.4-8          fastmap_1.0.1        rgeos_0.5-2         
[73] KernSmooth_2.23-16   stringi_1.4.6        parallel_4.0.0       crul_0.9.0          
[77] Rcpp_1.0.4.6         vctrs_0.2.4          leaflet_2.0.3 

> packageVersion("sen2r")
[1] ‘1.3.4’

> sen2r::load_binpaths()
$gdalinfo
[1] "C:\\OSGEO4~1\\bin\\gdalinfo.exe"

$python
[1] "C:\\OSGEO4~1\\bin\\python.exe"

$gdal_calc
[1] "C:\\OSGEO4~1\\bin\\python.exe C:\\OSGEO4~1\\bin\\GDAL_C~1.PY"

$gdal_fillnodata
[1] "C:\\OSGEO4~1\\bin\\python.exe C:\\OSGEO4~1\\bin\\GDAL_F~1.PY"

$sen2cor
[1] "C:\\Users\\usuario\\DOCUME~1\\R\\sen2cor\\SEN2CO~1.05-\\L2A_PR~1.BAT"

$aria2c
[1] "C:\\Users\\usuario\\DOCUME~1\\R\\aria2\\aria2c.exe"

attr(,"path")
[1] "C:\\Users\\usuario\\DOCUME~1\\SEN2R~1/paths.json"
[SQB_JSON.txt](https://github.com/ranghetti/sen2r/files/4567404/SQB_JSON.txt)

**Comment by ranghetti** [comment]
What error do you receive launching the following command in a cmd prompt?
```
C:\Users\usuario\DOCUME~1\R\sen2cor\SEN2CO~1.05-\L2A_PR~1.BAT C:\sentinel_and\restante\S2A_MSIL1C_20170722T153911_N0205_R011_T18TYS_20170722T154309.SAFE
```

Moreover, please verify and report the size and the content of the product `C:\sentinel_and\restante\S2A_MSIL1C_20170722T153911_N0205_R011_T18TYS_20170722T154309.SAFE`.

**Comment by vrgaliano** [comment]
> C:\Users\usuario\DOCUME~1\R\sen2cor\SEN2CO~1.05-\L2A_PR~1.BAT
Error: unexpected input in "C:\"
> C:\sentinel_and\restante\S2A_MSIL1C_20170722T153911_N0205_R011_T18TYS_20170722T154309.SAFE

The size of the product is 797MB. I am not too sure about the content, but the GRANULE folder seems to have all the bands.

**Comment by ranghetti** [comment]
Probably a bug on Windows: I added the use of `shortPathName()` on SAFE paths to solve previous problems with long paths and spaces, but actually the SAFE name can not be shortened.
I added a patch on a development branch: can you try if, after installing it, the problem will got solved?
```r
remotes::install_github("ranghetti/sen2r", ref = "hotfix/issue311")
```
(Before doing that, delete eventual "empty" L2A SAFE folders created by the function)

**Comment by vrgaliano** [comment]
It works now, thanks. It could be a bug on windows 10. Two other people
from my team had the same issue with different computers. All with windows
10. Just in case that helps.

**Comment by ranghetti** [comment]
The bug was fixed starting from GitHub version 1.3.4.9001. Until version 1.3.5 is released on CRAN, if other people of your team will need to perform the same operations, tell them to install the GitHub version with the following command:
```r
remotes::install_github("ranghetti/sen2r", ref = "hotfix/issue311")
```
